rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala

```diff
@@ -1,41 +1,70 @@
-
-    val listNamesObservable: Observable[String] = client.listDatabaseNames()
-
-    listNamesObservable.subscribe(new Observer[String] {
-
-      override def onError(e: Throwable): Unit = println(s"onError listNamesObservable: $e")
-      override def onComplete(): Unit = println("listNamesObservable onComplete")
-      override def onNext(result: String): Unit = println(s"onNext listNamesObservable: $result")
-    })
-
-    println("==========> BEFORE PUT")
-    println("==========> id")
-    println(id)
-    println("==========> value")
-    println(value)
-
-    val database = client.getDatabase("local")
-    val collection: MongoCollection[Document] = database.getCollection("test_collection")
-    Thread.sleep(5000)
-    val document: Document = Document("_id" -> id, "value" -> value)
-    val insertObservable: Observable[Completed] = collection.insertOne(document)
-    Thread.sleep(5000)
-    insertObservable.subscribe(new Observer[Completed] {
-      override def onNext(result: Completed): Unit = println(s"onNext insertObservable: $result")
-      override def onError(e: Throwable): Unit = println(s"onError insertObservable: $e")
-      override def onComplete(): Unit = println("insertObservable onComplete")
-    })
+  val client: MongoClient = MongoClient("mongodb://localhost:27017") //the actual host name of parent machine
+
+  val database = client.getDatabase("local")
+  val collection: MongoCollection[Document] = database.getCollection("test_collection")
+
+  def prepareDocument(matchToPersist: Match): Document = {
+    Document("_id" -> matchToPersist.matchId,
+      "radiantWin" -> matchToPersist.radiantWin,
+      "duration" -> matchToPersist.duration,
+      "startTime" -> matchToPersist.startTime,
+      "gameMode" -> matchToPersist.gameMode,
+      "direScore" -> matchToPersist.direScore,
+      "players" -> preparePlayersDocument(matchToPersist.players))
+  }
+
+  def preparePlayersDocument(players: List[Player]): List[Document] = {
+    var playersList = List[Document]()
+    for (player <- players) {
+      player.defineSide()
+      playersList = playersList :+ Document("accountId" -> player.accountId,
+        "heroId" -> player.heroId,
+        "kills" -> player.kills,
+        "deaths" -> player.deaths,
+        "assists" -> player.assists,
+        "denies" -> player.denies,
+        "goldPerMinute" -> player.goldPerMinute,
+        "xpPerMinute" -> player.xpPerMinute,
+        "level" -> player.level,
+        "heroDamage" -> player.heroDamage,
+        "heroHealing" -> player.heroHealing,
+        "goldCurrent" -> player.goldCurrent,
+        "goldSpent" -> player.goldSpent,
+        "isDire" -> player.isDire
+      )
+    }
+    playersList
+  }
+
+  def put(matchToPersist: Match): Unit = {
+    val findObservable = collection.find(equal("_id", matchToPersist.matchId)).first()
+
+    val foundDocument = Await.result(findObservable.toFuture(), Duration(10, TimeUnit.SECONDS))
```

Comment:
Maybe lets move 10 second to some configuration, so we could handle all timeout in a single place ?